packages/graphql/src/graphql/scalars/Duration.ts

```diff
@@ -39,41 +46,34 @@ export const SECONDS_PER_HOUR = SECONDS_PER_MINUTE * MINUTES_PER_HOUR;
 export const NANOSECONDS_PER_SECOND = 1000000000;
 
 export const parseDuration = (value: string) => {
-    const match = DURATION_REGEX.exec(value);
+    const matchIso = DURATION_REGEX_ISO.exec(value);
+    const matchDelimiter = DURATION_REGEX_WITH_DELIMITERS.exec(value);
+    const matchNoDelimiter = DURATION_REGEX_NO_DELIMITERS.exec(value);
 
+    const match = matchIso || matchDelimiter || matchNoDelimiter;
     if (!match) {
         throw new TypeError(`Value must be formatted as Duration: ${value}`);
     }
 
     const {
         negated,
-        // P[nY][nM][nD][T[nH][nM][nS]]
-        yearUnit = 0,
-        monthUnit = 0,
-        dayUnit = 0,
-        hourUnit = 0,
-        minuteUnit = 0,
-        secondUnit = 0,
-        // PnW
-        weekUnit = 0,
-        // PYYYYMMDDTHHMMSS | PYYYY-MM-DDTHH:MM:SS
-        yearDT = 0,
-        monthDT = 0,
-        dayDT = 0,
-        hourDT = 0,
-        minuteDT = 0,
-        secondDT = 0,
+        years: yearUnit = 0,
+        months: monthUnit = 0,
+        weeks: weekUnit = 0,
+        days: dayUnit = 0,
+        hours: hourUnit = 0,
+        minutes: minuteUnit = 0,
+        seconds: secondUnit = 0,
         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     } = match.groups!;
 
-    // NOTE: xUnit and xDT cannot both be nonzero by construction => (xUnit + xDT) = xUnit | xDT | 0
-    const years = +yearUnit + +yearDT;
-    const months = +monthUnit + +monthDT;
+    const years = +yearUnit;
```

Comment:
Maybe we can get rid of the leading sign?

```suggestion
    const years = yearUnit;
```

In fact, I think we may be able to get rid of this assignments all together

Reply:
That leading `+` is to cast a string to a number. That was the best way I could come up so far. Happy to hear a better approach!